Documentation/Coherency.md

```diff
@@ -0,0 +1,74 @@
+# Problem #
+-----------
+Official non-servicing builds of the .NET Core SDK are *coherent*, meaning that for each repo that contributes bits to the final product, exactly one version of that repo is used at all stages of the build.  For instance, CLI and core-setup both depend on CoreFX.  In a coherent build, this will be the same version of CoreFX, but in an incoherent build, CLI could have been built with a different version than core-setup.  This can cause problems which usually manifest in source-build as build errors along the lines of `Could not load file or assembly 'NuGet.Frameworks, Version=4.9.3.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The located assembly's manifest definition does not match the assembly reference.`, but could also result in runtime errors.  
+Source-build has traditionally worked around this by forcing all repos to use the same bits (the source-built ones) throughout the build process.  However, this brings in new problems:
+- We are not necessarily shipping the exact bits that an official release would.
```

Comment:
I think this is a bit understated. By forcing packages to build that shouldn't be building at that commit, we could be introducing new code that's (in the worst case) known to be broken. 

Covered inside https://github.com/dotnet/source-build/issues/923, but want to make sure this issue is visible. Maybe expand on it in the "cons" section below at least?

Reply:
Added stronger language in both places.